### PR TITLE
lenovo-dock: Add HP G2 miniport hub support

### PR DESCRIPTION
**Pull request by victor-cheng**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/lenovo-dock/lenovo-dock.quirk [hunk 1]
```diff
@@ -2,8 +2,15 @@
...
 [USB\VID_17EF&PID_30B5]
 Plugin = lenovo_dock
 GType = FuLenovoDockDmcDevice
 ParentGuid = USB\VID_17EF&PID_30B4
+
+[USB\VID_03F0&PID_0505]
+Plugin = lenovo_dock
+GType = FuLenovoDockMcuDevice
+Name = USB-C G2 Multiport Hub MCU Controller
+LenovoDockVersionFormat = HP
```
Comment (superm1) [inline plugins/lenovo-dock/lenovo-dock.quirk:16]: Lenovo format = HP?
That doesn't seem right.

Comment (superm1) [reply plugins/lenovo-dock/lenovo-dock.quirk:16]: Maybe the plugin is named wrong?

File: plugins/lenovo-dock/fu-lenovo-dock-mcu-device.c [hunk 2]
```diff
@@ -689,6 +730,35 @@ fu_lenovo_dock_mcu_device_set_progress(FuDevice *self, FuProgress *progress)
...
+fu_lenovo_dock_mcu_device_set_quirk_kv(FuDevice *device,
+				       const gchar *key,
+				       const gchar *value,
+				       GError **error)
+{
+	FuLenovoDockMcuDevice *self = FU_LENOVO_DOCK_MCU_DEVICE(device);
+	if (g_strcmp0(key, "LenovoDockVersionFormat") == 0) {
+		self->version_format = g_strdup(value);
+		return TRUE;
+	}
```
[the remark this answers is not in the document]
Comment (victor-cheng) [reply plugins/lenovo-dock/fu-lenovo-dock-mcu-device.c:743]: Currently, ODM/USI supports these two vendors to use USI-designed MCU update protocol, and supported devices with different vendors will be increased more. Do you think the plugin name should be re-named as USI-dock or something else?

[the remark this answers is not in the document]
Comment (hughsie) [reply plugins/lenovo-dock/fu-lenovo-dock-mcu-device.c:743]: Yes, I think usi-dock makes a lot more sense in this case.

**Review by hughsie — APPROVED** [review]

**Pull request merged**